From a7fb55633a2ba7030ebfb3f56ec6775160ff407f Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas@debian.org>
Date: Thu, 7 Jan 2021 09:42:40 +0100
Subject: [PATCH 19/21] Distinguish TARGET, DEB_TARGET_ARCH and
 DEB_TARGET_GNU_TYPE

TARGET is not overriden anymore, providing more flexibility for GNU
types unknown to dpkg-architecture.
---
 debian/rules | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/debian/rules b/debian/rules
index 032d08e..d59d9af 100755
--- a/debian/rules
+++ b/debian/rules
@@ -70,7 +70,7 @@ p_ctfn = libctf-nobfd0
 p_hst = $(p_bin)-for-host
 p_bld = $(p_bin)-for-build
 
-p_cross = $(p_bin)-$(subst _,-,$(TARGET))
+p_cross = $(p_bin)-$(subst _,-,$(DEB_TARGET_GNU_TYPE))
 
 # version used for p_bld/p_hst dependencies
 min_ver = 2.30-2
@@ -240,12 +240,14 @@ ifneq (,$(TARGET))
   # and as the target name (e.g. arm-linux-gnu).
   try_convert := $(shell dpkg-architecture -f -a$(TARGET) -qDEB_HOST_GNU_TYPE 2>/dev/null)
   ifneq ($(try_convert),)
-    override TARGET := $(try_convert)
+    DEB_TARGET_GNU_TYPE := $(try_convert)
+  else
+    DEB_TARGET_GNU_TYPE = $(TARGET)
   endif
-  DPKG_TARGET_VARS := $(shell dpkg-architecture -f -t$(TARGET))
+  DPKG_TARGET_VARS := $(shell dpkg-architecture -f -t$(DEB_TARGET_GNU_TYPE))
   DEB_TARGET_MULTIARCH = $(call vafilt,$(DPKG_TARGET_VARS),DEB_HOST_MULTIARCH)
   DEB_TARGET_ARCH      = $(call vafilt,$(DPKG_TARGET_VARS),DEB_HOST_ARCH)
-  for_target           = dpkg-architecture -f -t$(TARGET) -c
+  for_target           = dpkg-architecture -f -t$(DEB_TARGET_GNU_TYPE) -c
 else
   for_target           =
 endif
@@ -590,7 +592,7 @@ stamps/control: debian/control.in debian/control.cross.in
 ifneq (,$(TARGET))
 	sed -n -e 's/@dpkg_dev@/$(DPKG_DEV)/;/^Package: binutils-for-build$$/q;p' \
 		< debian/control.in > debian/control
-	sed -e "s/@target@/$$(echo -n $(TARGET) | sed s/_/-/g)/g" \
+	sed -e "s/@target@/$(subst _,-,$(DEB_TARGET_GNU_TYPE))/g" \
 		-e "s/@host_archs@/any/" \
 		-e 's/@binutils_alt_triplet@//' \
 		< debian/control.cross.in >> debian/control
@@ -976,7 +978,7 @@ ifeq ($(with_cross),yes)
        build_stamps += $(addprefix stamps/build.,$(CROSS_ARCHS))
 endif
 ifneq (,$(TARGET))
-   ifeq ($(TARGET),hppa64-linux-gnu)
+   ifeq ($(DEB_TARGET_GNU_TYPE),hppa64-linux-gnu)
      build_stamps = stamps/build-hppa64
    else
      build_stamps = stamps/build.$(DEB_TARGET_ARCH)
@@ -1008,7 +1010,7 @@ ifeq ($(with_cross),yes)
 endif
 
 ifneq (,$(TARGET))
-    ifeq ($(TARGET),hppa64-linux-gnu)
+    ifeq ($(DEB_TARGET_GNU_TYPE),hppa64-linux-gnu)
       install_stamp = stamps/install-hppa64
     else
       install_stamp = stamps/install.$(DEB_TARGET_ARCH)
-- 
2.30.1

